boards/arm/stm32f0_disco/Kconfig.defconfig

```diff
@@ -0,0 +1,33 @@
+# Kconfig - STM32F0 development board with STM32F051R8T6 MCU
+#
+# Copyright (c) 2018 Nathan Tsoi <nathan@vertile.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+if BOARD_STM32F0_DISCO
+
+config BOARD
+	default stm32f0_disco
+
+if UART_CONSOLE
+
+config UART_STM32_PORT_1
+	default y
+
+endif # UART_CONSOLE
+
+# Disable ports D and F to allow kernel test cases common and
+# contex to run on stm32f0_disco.
+# Test cases fail due to 'SRAM' region overflow.
+if GPIO
+
+config GPIO_STM32_PORTD
```

Comment:
Thanks to move GPIO definition in arch/arm/soc/st_stm32/stm32f0/Kconfig.defconfig.stm32f051x8.
Then disable in stm32f0_disco_defconfig

Reply:
Thx for the feedback @erwango. I'm not sure I follow though. I think you're asking for something like:

```
diff --git a/boards/arm/stm32f0_disco/Kconfig.defconfig b/boards/arm/stm32f0_disco/Kconfig.defconfig
index 91ccdeda9..f2c417b3a 100644
--- a/boards/arm/stm32f0_disco/Kconfig.defconfig
+++ b/boards/arm/stm32f0_disco/Kconfig.defconfig
@@ -17,17 +17,4 @@ config UART_STM32_PORT_1

 endif # UART_CONSOLE

-# Disable ports D and F to allow kernel test cases common and
-# contex to run on stm32f0_disco.
-# Test cases fail due to 'SRAM' region overflow.
-if GPIO
-
-config GPIO_STM32_PORTD
-       default n
-
-config GPIO_STM32_PORTF
-       default n
-
-endif # GPIO
-
 endif # BOARD_STM32F0_DISCO
diff --git a/boards/arm/stm32f0_disco/stm32f0_disco_defconfig b/boards/arm/stm32f0_disco/stm32f0_disco_defconfig
index b64c9eb14..25dd94a10 100644
--- a/boards/arm/stm32f0_disco/stm32f0_disco_defconfig
+++ b/boards/arm/stm32f0_disco/stm32f0_disco_defconfig
@@ -30,6 +30,8 @@ CONFIG_PINMUX=y

 # GPIO Controller
 CONFIG_GPIO=y
+CONFIG_GPIO_STM32_PORTD=n
+CONFIG_GPIO_STM32_PORTF=n

 # Clock configuration
 CONFIG_CLOCK_CONTROL=y
```

Does that look close?

Reply:
@nathantsoi : Exactly